wagtail/models/__init__.py

```diff
@@ -2896,15 +2862,6 @@ def save(self, user=None, *args, **kwargs):
     def as_object(self):
         return self.content_object.with_content_json(self.content)
 
-    def as_page_object(self):
```

Comment:
Shall we remove the admonition from the docs as well?

https://github.com/wagtail/wagtail/blob/b336536015da7c3ecf1c5249e4fb3cc8062ca9d7/docs/reference/pages/model_reference.md?plain=1#L801-L803

Not sure how long we should keep these, but we also still have 3.0 stuff as well that should probably be removed.